usermatch/src/App.js

```diff
@@ -1,24 +1,87 @@
-import logo from './logo.svg';
+import React, { useState, useEffect } from 'react';
 import './App.css';
 
 function App() {
+  const [location, setLocation] = useState(null);
+
+  function getLocation () {
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition(success, error);
+    } else {
+      console.log("Geolocation not supported");
+    }
+  }
+    
+  function success(position) {
+    const latitude = position.coords.latitude;
+    const longitude = position.coords.longitude;
+    setLocation({ latitude, longitude });
+    
+    fetch('http://localhost:3000/updateLocation')
+      .then((res) => res.json())
+      .then((data) => setLocation(data))
+      .catch((error) => console.error("Error fetching location: ", error));
+  }
+
+  function error() {
+    console.log("Unable to retrieve your location");
+  }
+
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <main className="app-main">
+      <div className="location-button">
+        <button onClick={getLocation}>Allow Location</button>
+      </div>
+      <div>
+        <profileInput/>
+      </div>
+  </main>
+  );
+}
+
+function profileInput() {
+  const setProfile = Array.from(document.querySelectorAll('#userProfile input')).reduce((acc, input) => ({...acc, [input.id]: input.value }), {});
+  
+  fetch('http://localhost:3000/userprofile')
```

Comment:
We want this to be a POST request because we are sending data _TO_ the server (backend)
Perhaps something  like
```
    fetch('http://localhost:3000/updateLocation', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        latitude: latitude,
        longitude: longitude,
      }),
    }).catch((error) => console.error("Error fetching location: ", error));